wcfsetup/install/files/lib/system/cache/tolerant/TagCloudCache.class.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace wcf\system\cache\tolerant;
+
+use wcf\data\tag\Tag;
+use wcf\data\tag\TagCloudTag;
+use wcf\system\database\util\PreparedStatementConditionBuilder;
+use wcf\system\WCF;
+
+/**
+ * Caches the tag cloud.
+ *
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ *
+ * @extends AbstractTolerantCache<array<string, TagCloudTag>>
+ */
+class TagCloudCache extends AbstractTolerantCache
+{
+    public function __construct(
+        /** @var int[] */
+        public readonly array $objectTypeIDs,
+        /** @var int[] */
+        public readonly array $languageIDs = [],
+    ) {
+    }
+
+    private static function compareTags(TagCloudTag $tagA, TagCloudTag $tagB): int
+    {
+        if ($tagA->counter > $tagB->counter) {
+            return -1;
+        }
+        if ($tagA->counter < $tagB->counter) {
+            return 1;
+        }
+
+        return 0;
+    }
+
+    #[\Override]
+    public function getLifetime(): int
+    {
+        return 3_600;
+    }
+
+    #[\Override]
+    protected function rebuildCacheData(): array
+    {
+        if ($this->objectTypeIDs === []) {
+            return [];
+        }
+
+        return $this->fetchTags($this->fetchTagIDs());
+    }
+
+    /**
+     * @param list<int> $tagIDs
+     *
+     * @return array<string, TagCloudTag>
+     */
+    protected function fetchTags(array $tagIDs): array
+    {
+        if ($tagIDs === []) {
+            return [];
+        }
+
+        $conditionBuilder = new PreparedStatementConditionBuilder();
+        $conditionBuilder->add('tagID IN (?)', [\array_keys($tagIDs)]);
+        $sql = "SELECT *
+                FROM   wcf1_tag
+                " . $conditionBuilder;
+        $statement = WCF::getDB()->prepare($sql);
+        $statement->execute($conditionBuilder->getParameters());
+
+        $tags = [];
+        while ($row = $statement->fetchArray()) {
+            $row['counter'] = $tagIDs[$row['tagID']];
+            $tags[$row['name']] = new TagCloudTag(new Tag(null, $row));
+        }
+
+        // sort by counter
+        \uasort($tags, self::compareTags(...));
```

Comment:
If you need a comment for this, then the name of the method is poorly chosen. Maybe rename it to `sortTagsByCounter()` and get rid of the comment?